test/jsonldparser.spec.js

```diff
@@ -0,0 +1,115 @@
+import JsonLDParser from '../src/parser/jsonldparser';
+import responseWithoutEmbbeded from './fixtures/response-without-embbeded';
+import responseWithEmbbeded from './fixtures/response-with-embbeded';
+import responseWithOperations from './fixtures/response-with-operations';
+import Relation from '../src/model/relation';
+import Thing from '../src/model/thing';
+
+describe('JsonLDParser withouth embedded', () => {
+	it('should parse the thing', () => {
+		const parser = new JsonLDParser();
+
+		const {thing, _} = parser.parseThing(responseWithoutEmbbeded);
+
+		expect(thing.id).toBe('https://apiosample.wedeploy.io/p/people/9');
+		expect(thing.types).toEqual(['Person']);
+		expect(thing.attributes.name).toBe('Loy Hermiston');
+	});
+
+	it('should parse nested array', () => {
+		const parser = new JsonLDParser();
+
+		const {thing, _} = parser.parseThing(responseWithoutEmbbeded);
+
+		expect(thing.attributes.jobTitle).toEqual([
+			'Dynamic Tactics Orchestrator',
+		]);
+	});
+
+	it('should parse nested objects', () => {
+		const parser = new JsonLDParser();
+
+		const {thing, _} = parser.parseThing(responseWithoutEmbbeded);
+		const address = thing.attributes.address;
+
+		expect(address.streetAddress).toBe('1228 Ada Shoal');
+	});
+});
+
+describe('JsonParser with embbeded things', () => {
+	it('should replace id with relation', () => {
+		const parser = new JsonLDParser();
+
+		const {thing, _} = parser.parseThing(responseWithEmbbeded);
+
+		const view = thing.attributes.view;
+
+		expect(view instanceof Relation).toBeTruthy();
+	});
+
+	it('should replace embedded thing with relation', () => {
+		const parser = new JsonLDParser();
+
+		const {thing, _} = parser.parseThing(responseWithEmbbeded);
+
+		const view = thing.attributes.view;
+		const firstMember = thing.attributes.member[0];
+
+		expect(view instanceof Relation).toBeTruthy();
```

Comment:
this is a bit strange in javascript but Ok (duck typing)

Reply:
I agree... I have to think a better way to model this in js